### PR TITLE
remove scalding-jbdc

### PR DESCRIPTION
**Pull request by hazel-sudz**
based on https://github.com/twitter/scalding/issues/1968. We are not aware of any popular use of this module. Even if someone is using it it will likely need to be re-written in future scalding releases because: 

1. it is hard-wired to cascading 
2. it depends on cascading ecosystem packages that have not been maintained since cascading 2.X release

I would prefer to reduce support burden on things that do not provide value to most consumers. This is also something that can implemented on the consumer side if it is really required.

### CODE REVIEW

**Comment by hazel-sudz** [comment]
I found `com.twitter.scalding.db` and looks like in the past there were plans to move jbdc in there but there was never any incentive to do so after a while.

**Comment by hazel-sudz** [comment]
based on https://github.com/twitter/scalding/pull/1382 twitter used/uses an internal implementation of jbdc for scalding

**Comment by tlazaro** [comment]
I don't remember, if jdbc was used at Twitter. `scalding-db` for sure was. If @germanschnyder or @nownikhil could chime in.

**Comment by johnynek** [comment]
we can revert this if Twitter corrects us later. Let's keep scalding-db for now.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 name: CI
 
-# run on all branches
 on:
   pull_request:
     branches:
       - '**'
+  # it's annoying to have CI double run on PR/push
+  # limit push to only final integration branches like develop
```
Comment (germanschnyder) [inline .github/workflows/CI.yml:8]: Agree

**Comment by nownikhil** [comment]
Sorry for the delay, was out for few days. Verified that its not being used at Twitter. Thanks @daniel-sudz

**Pull request merged**